### ISSUE TITLE
Timber 1.9.5 breaks my site "Ce site rencontre des difficultés techniques."

### ISSUE THREAD

**Issue by AL20** [timber/timber#2020]
Hi,

I just updated my website with Timber 1.9.5 and I get this message '`Ce site rencontre des difficultés techniques`.' (in english : _The site is experiencing technical difficulties_. equivalent to the White Screen of Death since WP5.2) on all the pages that use Timber + Twig templates.

My setup (with a custom theme) :
- WP last version (5.2.1)
- ACF Pro (last version)
- CPT UI (last version)
- Woocommerce (last version)
- WPML (last version)

I tried to desactivate them individually but the issue remains.

In advance, thanks for your help.

**Comment by palmiak** [comment]
This message is just more user friendly 500 in WP 5.2

If you would downgrade timber to earlier version will the problem still
exists?

Also try to turn on WP_DEBUG it should show you the real problem

pon., 3 cze 2019, 12:17 użytkownik AL20 <notifications@github.com> napisał:

**Comment by AL20** [comment]
When I rollback to Timber 1.9.4, the issue is gone.
I set :
```
define('WP_DEBUG', true);
define('WP_DEBUG_LOG', true);
define('WP_DEBUG_DISPLAY', true);
```
but I don't get more informations.

**Comment by palmiak** [comment]
I see you are using WPML. From what I remember Wpml also uses twig and there might be some conflict here. If I would have to guess I think it might be related with this update - https://github.com/timber/timber/pull/1963 . 

Is there anything in an error_log?

**Comment by AL20** [comment]
Maybe, I know WPML brings a lot of issue in general…
Here's what I got from the log :

```
Warning: Cannot declare interface Twig_LoaderInterface, because the name is already in use in /Users/vinsan/Dropbox/DEV/Sites/site_skyfarms_boutique/wp-content/plugins/timber-library/vendor/twig/twig/src/Loader/LoaderInterface.php on line 61

Warning: Cannot declare interface Twig_ExistsLoaderInterface, because the name is already in use in /Users/vinsan/Dropbox/DEV/Sites/site_skyfarms_boutique/wp-content/plugins/timber-library/vendor/twig/twig/src/Loader/ExistsLoaderInterface.php on line 33

Warning: Cannot declare interface Twig_SourceContextLoaderInterface, because the name is already in use in /Users/vinsan/Dropbox/DEV/Sites/site_skyfarms_boutique/wp-content/plugins/timber-library/vendor/twig/twig/src/Loader/SourceContextLoaderInterface.php on line 38

Warning: Cannot declare class Twig_Loader_Filesystem, because the name is already in use in /Users/vinsan/Dropbox/DEV/Sites/site_skyfarms_boutique/wp-content/plugins/timber-library/vendor/twig/twig/src/Loader/FilesystemLoader.php on line 324

Warning: Cannot declare class Twig_Environment, because the name is already in use in /Users/vinsan/Dropbox/DEV/Sites/site_skyfarms_boutique/wp-content/plugins/timber-library/vendor/twig/twig/src/Environment.php on line 1638

Warning: Cannot declare interface Twig_CacheInterface, because the name is already in use in /Users/vinsan/Dropbox/DEV/Sites/site_skyfarms_boutique/wp-content/plugins/timber-library/vendor/twig/twig/src/Cache/CacheInterface.php on line 60

Warning: Cannot declare class Twig_Cache_Null, because the name is already in use in /Users/vinsan/Dropbox/DEV/Sites/site_skyfarms_boutique/wp-content/plugins/timber-library/vendor/twig/twig/src/Cache/NullCache.php on line 42

Fatal error: Cannot redeclare twig_cycle() (previously declared in /Users/vinsan/Dropbox/DEV/Sites/site_skyfarms_boutique/wp-content/plugins/woocommerce-multilingual/vendor/twig/twig/lib/Twig/Extension/Core.php:288) in /Users/vinsan/Dropbox/DEV/Sites/site_skyfarms_boutique/wp-content/plugins/timber-library/vendor/twig/twig/src/Extension/CoreExtension.php on line 332

```

**Comment by jarednova** [comment]
Yeah, it looks like it could be because of a conflict with WPML's version of Twig. We upgraded the Twig version that 1.9.5 ships with — let me see if there's a way I can investigate that

**Comment by jarednova** [comment]
@AL20 I'm not able to trigger the error you reported, but I'm pretty confident this is an issue with a collision of Twig versions. Can you replace your `wp-content/plugins/timber-library` directory with this test version and see if that resolves the issue?

https://www.zipshare.com/download/eyJhcmNoaXZlSWQiOiI3YzVhMjY0NS0yNGQwLTQ3NjQtYmQyZC1mMWQxNjBkNTYzMzYiLCJlbWFpbCI6ImphcmVkQHVwc3RhdGVtZW50LmNvbSJ9

**Comment by AL20** [comment]
I use your test version and ahve a new error : 
`Fatal error: Uncaught Error: Class 'Twig\Loader\FilesystemLoader' not found in /Users/vinsan/Dropbox/DEV/Sites/site_skyfarms_boutique/wp-content/plugins/timber-library/lib/Loader.php:144 Stack trace: #0 /Users/vinsan/Dropbox/DEV/Sites/site_skyfarms_boutique/wp-content/plugins/timber-library/lib/Loader.php(102): Timber\Loader->get_loader() #1 /Users/vinsan/Dropbox/DEV/Sites/site_skyfarms_boutique/wp-content/plugins/timber-library/lib/Timber.php(309): Timber\Loader->choose_template(Array) #2 /Users/vinsan/Dropbox/DEV/Sites/site_skyfarms_boutique/wp-content/plugins/timber-library/lib/Timber.php(377): Timber\Timber::compile('pages/page-home...', Array, false, 'default', true) #3 /Users/vinsan/Dropbox/DEV/Sites/site_skyfarms_boutique/wp-content/plugins/timber-library/lib/Timber.php(404): Timber\Timber::fetch('pages/page-home...', Array, false, 'default') #4 /Users/vinsan/Dropbox/DEV/Sites/site_skyfarms_boutique/wp-content/themes/skyfarms/home.php(40): Timber\Timber::render('pages/page-home...', Array) #5 /Users/vinsan/Dropbox/ in /Users/vinsan/Dropbox/DEV/Sites/site_skyfarms_boutique/wp-content/plugins/timber-library/lib/Loader.php on line 144`

**Comment by jarednova** [comment]
Crap. Well at least it's _new_. Can you deactivate plugins and isolate which one causes the issue? For example, if it's just WPML that'd be helpful so I can try to simulate the issue on my end

**Comment by windycitymoon** [comment]
Just thought I'd chime in on this one...

Ran core & plugin updates last night on 14 dev sites, all of the ones using Timber are WSOD right now.

Running  WP 5.2.1
For this site, all plugins disabled except for Timber.
I roll back to Timber 1.9.4 all is fine....Error makes more sense if Twig got updated.

Fatal error: Uncaught Twig\Error\LoaderError: Unable to find template "@components/.twig" (looked into: /app/wp-content/themes/gesso/pattern-lab/source/_patterns/03-components) in "@components/page-builder/page-builder.twig" at line 3. in /app/wp-content/plugins/timber-library/vendor/twig/twig/src/Loader/FilesystemLoader.php:266 Stack trace: #0 /app/wp-content/plugins/timber-library/vendor/twig/twig/src/Loader/FilesystemLoader.php(161): Twig\Loader\FilesystemLoader->findTemplate('@components/.tw...') #1 /app/wp-content/plugins/timber-library/vendor/twig/twig/src/Environment.php(351): Twig\Loader\FilesystemLoader->getCacheKey('@components//.t...') #2 /app/wp-content/plugins/timber-library/vendor/twig/twig/src/Environment.php(445): Twig\Environment->getTemplateClass('@components//.t...') #3 /app/wp-content/plugins/timber-library/vendor/twig/twig/src/Template.php(369): Twig\Environment->loadTemplate('@components//.t...', NULL) #4 /app/wp-content/plugins/timber-library/vendor/twig/twig/src/Environment.php(497) : eval()'d code in /app/wp-content/plugins/timber-library/vendor/twig/twig/src/Loader/FilesystemLoader.php on line 266

**Comment by windycitymoon** [comment]
Hey @jarednova - any plans to release a hotfix to the one-click version of this on WP, or should we just be rolling back and holding on for a 1.9.6 version.

**Comment by palmiak** [comment]
https://github.com/timber/timber/issues/2016#issuecomment-498838477 here is your answer

**Comment by ivillacreative** [comment]
Not Sure if this should be a new issue or If I can tack onto this one, But I've also found that 1.9.5 Breaks a part of my site. it has a conflict with Paid Membership pro directory. 

I looked through the other plugin and don't see the use of any twig files so I don't think it could be that.

What makes this error hard for me to debug is that WP debug spits out no error, just a white screen.

For now, I will just roll back to fix the issues

**Comment by palmiak** [comment]
Hi and can you check your error logs? There should be some info there.

pt., 7 cze 2019, 18:31 użytkownik Ivan Villa <notifications@github.com>
napisał:

**Comment by jarednova** [comment]
@ivillacreative thanks for the heads-up on that. Yes, please post the logs so we can see what's triggering that. I'm also in the midst of a release to pull 1.9.5 and hold 1.9.x users at an earlier version while we smooth out upgrade issues

**Comment by ivillacreative** [comment]
I wish I could, In this case, both the logs and page come out completely empty with no errors. It's a very odd Issue, I've been chatting with support at Paid Membership pro and they say they don't use twig so They don't believe its that for them. 

sorry I can't be more help.

**Comment by tomasvanrijsse** [comment]
I've this error on a site as well (running on Timber version 1.10.0)
`PHP Warning:  Cannot declare interface Twig_CacheInterface, because the name is already in use`

I've done a `grep Twig_CacheInterface ./* -R` and found that it's running on two locations.

```
./vendor/twig/twig/lib/Twig/CacheInterface.php
./public/wp-content/plugins/timber-library/vendor/twig/twig/lib/Twig/CacheInterface.php
```
The twig libary in the vendor directory is installed using composer for [Themosis Framework](https://framework.themosis.com/) v1.3.3
After downgrading Timber to v1.9.4 the issues went away.

**Comment by jarednova** [comment]
Thanks @tomasvanrijsse — good to know about that conflict. 

We've pulled 1.9.5 due to the compatibility issues like this one. I'm considering pulling 1.10.* from WordPress.org too. I'm wondering (out loud here) if the conflict risk outweighs the benefit from 1.10 vs. 1.9. Perhaps its better to leave things at a lower version for plugin/wp.org users — whereas Composer users can continue to upgrade (since composer can better manage versions/conflicts)

**Comment by jarednova** [comment]
I posted this in other places, but just to make sure it's clear:

>So after further investigation, it seems that the issues here are overlapping but down to a central problem: WPML's Twig conflicts with Timber's Twig. From my testing, the issue only manifests itself with the WooCommerce and WooCommerce Multilingual plugin running.
Unfortunately, attempting to rollback our Twig version (which seems to solve the issue under some limited circumstances) creates a chain of other issues that need to be resolved.
I'm going to release 1.10.0 as a version that does NOT work with these plugins (at least, for the time being). I'll note in the release notes to hopefully prevent developers from upgrading and then getting frustrated. So what should you/they do? I recommend using 1.9.2 — this seems to be stable before these various WPML / WooCommerce issues started. Once 1.10 is released I'm going to pull 1.9.5, 1.9.4 and 1.9.3. The only thing they seem to have done is introduce a raft of compatibility issues for sites.
We're in talks via email with the WPML team to hopefully figure out a long-term solution to this for 1.10 and beyond

A new release of WPML is coming soon which will resolve the conflicts. I'm in touch with their team and will post an update once I've verified the fix
